src/conn/conn_dhandle.c

```diff
@@ -332,6 +330,16 @@ __wt_conn_btree_open(
 	/* Set any special flags on the handle. */
 	F_SET(btree, LF_MASK(WT_BTREE_SPECIAL_FLAGS));
 
+	/*
+	 * Allocate data-source statistics memory. We don't allocate that memory
+	 * when allocating the data-handle because not all data handles need
+	 * statistics (for example, handles used for checkpoint locking). If we
+	 * are reopening the handle, then it may already have statistics memory,
+	 * check to avoid the leak.
+	 */
+	if (!F_ISSET(dhandle, WT_DHANDLE_OPEN))
```

Comment:
It probably isn't necessary to check the `WT_DHANDLE_OPEN` flag here, or at least something is wrong if we get here and the handle is already open.  That said, I don't mind it for clarity.